packages/xo-web/src/common/custom-fields.js

```diff
@@ -70,19 +121,53 @@ const CustomFields = decorate([
   }),
   provideState({
     effects: {
-      addCustomField: () => (state, { object: { id } }) =>
-        form({
-          render: props => <AddCustomFieldModal {...props} />,
-          defaultValue: { name: '', value: '' },
+      addCustomField: () => (state, { object: { id } }) => {
+        const dateTime = moment()
+        return form({
+          component: CustomFieldModal,
+          defaultValue: {
+            date: dateTime.format(DATE_FORMAT),
+            isDate: false,
+            name: '',
+            text: '',
+            time: dateTime.format(TIME_FORMAT),
+          },
           header: (
             <span>
               <Icon icon='add' /> {_('addCustomField')}
             </span>
           ),
-        }).then(({ name, value }) => addCustomField(id, name.trim(), value.trim())),
+        }).then(({ date, isDate, name, text, time }) =>
+          addCustomField(id, name.trim(), isDate ? `${date} ${time}Z` : text.trim())
+        )
+      },
       removeCustomField: (_, { currentTarget: { dataset } }) => (_, { object: { id } }) =>
         removeCustomField(id, dataset.name),
-      setCustomFieldValue: (_, value, { name }) => (_, { object: { id } }) => setCustomField(id, name, value),
+      setCustomField: (effects, { name, value }) => (state, { object: { id } }) => {
+        const isDate = !isInvalidDate(value)
+        return form({
+          render: props => <CustomFieldModal {...props} update />,
+          defaultValue: isDate
+            ? {
+                date: moment(value).format(DATE_FORMAT),
+                isDate,
+                name,
+                time: moment(value).format(TIME_FORMAT),
```

Comment:
Won't this be timezoned?

Reply:
Yes. See #4730 description
`other-config (MRW): XenCenter.CustomFields.mydatefield1: 2020-11-06 13:50:54Z;`